tests/framework/data/ActiveDataProviderTest.php

```diff
@@ -101,6 +101,21 @@ public function testActiveRelationVia()
         $this->assertCount(2, $items);
     }
 
+    public function testActiveRelationViaHasRelation()
```

Comment:
a new method is introduced in framework/db/BaseActiveRecord.php but a unit test is supplied in ActiveDataProviderTest.php. I'd expect specific unit tests for BaseActiveRecord?

Reply:
tests/framework/db/ActiveRecordTest.php
```php
    public function testJoinWithViaRelation()
    {
        Order::getDb()->getQueryBuilder()->separator = "\n";
        $rows = Order::find()->joinWith([
            'items2' => function ($q) {
                $q->orderBy('item.id');
            },
        ])->all();
        $this->assertNotEmpty($rows);
    }
```